lxd/response/response.go

```diff
@@ -200,19 +203,29 @@ func (r *syncResponse) Render(w http.ResponseWriter, req *http.Request) error {
 		}
 	}
 
+	defer func() {
+		if err == nil {
```

Comment:
You're creating an err var at the start of the function by using `(err error) ` as the return type, which is equivalent to doing `var err error` as the first line of the function.

However on line 200 a new err var is created within the if statement starting line 195 and then returned as a `SmartError(err).Render(...)` and im not sure if that is going to have set the top level err that is being consulted in this defer statement.

Reply:
Is this not relevant because its not returning a "render error" but instead a "response error"?

Reply:
This error should not impact the calling of the callback function since it is before the deferring of said call, so if it is not nil the callback function will not be called here, but on `SmartError(err).Render` instead. I placed a comment making this clearer.